### PR TITLE
[AppConfig] Not share SyncTokenPolicy across Configuration clients

### PR DESCRIPTION
**Pull request by mssfang**
# Description

SyncTokenPolicy should be exist as a single standalone object per Configuration client but not shared cross clients.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [X] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [X] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [X] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClientBuilder.java [hunk 1]
```diff
@@ -186,7 +185,7 @@ public ConfigurationClientBuilder() {
      * and {@link #retryPolicy(HttpPipelinePolicy)} have been set.
      */
     public ConfigurationClient buildClient() {
-        return new ConfigurationClient(buildInnerClient(), DEFAULT_SYNC_TOKEN_POLICY);
+        return new ConfigurationClient(buildInnerClient(), new SyncTokenPolicy());
```
Comment (alzimmermsft) [inline sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClientBuilder.java:188]: Does the `SyncTokenPolicy` used in the pipeline and the one passed to the ConigurationClient need to be the same?

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClientBuilder.java [hunk 2]
```diff
@@ -117,7 +117,6 @@ public final class ConfigurationClientBuilder implements
     ConfigurationTrait<ConfigurationClientBuilder>,
     EndpointTrait<ConfigurationClientBuilder> {
     private static final RetryPolicy DEFAULT_RETRY_POLICY = new RetryPolicy("retry-after-ms", ChronoUnit.MILLIS);
```
Comment (mrm9084) [inline sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClientBuilder.java:119]: This is also going to be an issue if the retry policy is changed between build operations.

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClientBuilder.java [hunk 3]
```diff
@@ -117,7 +117,6 @@ public final class ConfigurationClientBuilder implements
...
-    private static final SyncTokenPolicy DEFAULT_SYNC_TOKEN_POLICY = new SyncTokenPolicy();
```
Comment (mrm9084) [inline sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClientBuilder.java:120]: It's not letting me comment on the lines but could perCallPolicies and perRetryPolicies being shared ever cause and issue?

**Review by mrm9084 — CHANGES_REQUESTED** [review]

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClientBuilder.java (hunk 1 above)
Comment (mssfang) [reply sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClientBuilder.java]: SyncTokenPolicy needs to be the same object.  Updated.

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClientBuilder.java (hunk 3 above)
Comment (mssfang) [reply sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClientBuilder.java:119]: `perCallPolicies`  and `perRetryPolicies` are shared and can be updated per client builder.  There is a method
```
public ConfigurationClientBuilder addPolicy(HttpPipelinePolicy policy)
```
So we need to keep it as shared per builder.

**Review by alzimmermsft — APPROVED** [review]
The changes for SyncTokenPolicy look good

**Review by mrm9084 — APPROVED** [review]

**Pull request merged**